homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -103,22 +115,43 @@ def create_powerview_shade_entity(
 ):
     """Create a PowerViewShade entity."""
 
+    # order here is important as both ShadeTDBU are listed in aiovapi as can_tilt
+    # and both require their own class here to work
+    if isinstance(shade, ShadeTdbu):
+        return [
+            PowerViewShadeTDBUTop(
+                coordinator, device_info, room_name, shade, name_before_refresh
+            ),
+            PowerViewShadeTDBUBottom(
+                coordinator, device_info, room_name, shade, name_before_refresh
+            ),
+        ]
+
     if isinstance(shade, Silhouette):
-        return PowerViewShadeSilhouette(
-            coordinator, device_info, room_name, shade, name_before_refresh
-        )
+        return [
+            PowerViewShadeSilhouette(
+                coordinator, device_info, room_name, shade, name_before_refresh
+            )
+        ]
 
-    return PowerViewShade(
-        coordinator, device_info, room_name, shade, name_before_refresh
-    )
+    if shade.can_tilt:
+        return [
+            PowerViewShadeWithTilt(
+                coordinator, device_info, room_name, shade, name_before_refresh
+            )
+        ]
+
+    return [
+        PowerViewShade(coordinator, device_info, room_name, shade, name_before_refresh)
+    ]
```

Comment:
This is getting a bit complex.  Since everything takes the same arguments, can you make the api module to hass entity a mapping?

Reply:
i dont think there is any other type to be mapped to be honest - didnt we move to this function in place of a mapping?

Reply:
There was only a single entity type before

Reply:
yep that did the trick